core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputReader.java

```diff
@@ -266,23 +278,26 @@ public void transform(KeyValue<LongWritable, Object> input, Emitter<StructuredRe
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
 
-    excelInputreaderConfig.validate();
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    excelInputreaderConfig.validate(collector);
 
     if (Strings.isNullOrEmpty(excelInputreaderConfig.columnList) &&
       Strings.isNullOrEmpty(excelInputreaderConfig.outputSchema)) {
-      throw new IllegalArgumentException("'Field Name Schema Type Mapping' input cannot be empty when the empty " +
-                                           "input value of 'Columns To Be Extracted' is provided.");
+      collector.addFailure(
+          "'Field Name Schema Type Mapping' and 'Columns To Be Extracted' cannot both be empty.", null)
+          .withConfigProperty(OUTPUT_SCHEMA).withConfigProperty(COLUMN_LIST);
     }
 
     createDatasets(pipelineConfigurer, null);
-    init();
+    init(collector);
     getOutputSchema();
     pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
   }
 
   @Override
   public void prepareRun(BatchSourceContext batchSourceContext) throws Exception {
-    excelInputreaderConfig.validate();
+    excelInputreaderConfig.validate(batchSourceContext.getFailureCollector());
```

Comment:
call getOrThrowException after this method call

Reply:
Fixed